src/workflows/dags/forecast_site_quarterly_training.py

```
@@ -0,0 +1,295 @@
+    start_date_str = DateUtils.date_to_str(start_date, str_format="%Y-%m-%d")
+    end_date_str = DateUtils.date_to_str(current_date, str_format="%Y-%m-%d")
+
+    raw_data = BigQueryApi().fetch_site_data_for_forecast_jobs(
+        start_date_time=start_date_str,
+        end_date_time=end_date_str,
+        job_type="train",
+        min_hours=18,
+    )
+
+    if raw_data.empty:
+        raise ValueError("No site forecast training data found in the selected period.")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)